src/components/DonationCard/DonationCard.tsx

```diff
@@ -1,23 +1,68 @@
 import styles from './DonationCard.module.css';
 
-export interface DonationCardProps {
+interface DonationCardProps {
    title: string;
    description: string;
    donationTarget: number;
+   amount: number;
+   isOpen: boolean;
+   onShowMore: () => void;
 }
 
-function DonationCard({ title, description, donationTarget }: DonationCardProps) {
+const getProgressClassNameByProgress = (progress: number) =>
+   progress < 33
+      ? styles.progressLow
+      : progress < 67
+        ? styles.progressMedium
+        : styles.progressHigh;
+
+const getButtonClassNames = (progress: number) =>
+   `${styles.button} ${getProgressClassNameByProgress(progress)}`;
+
+const getDonationCardProgressClassNames = (progress: number) =>
+   `${styles.progress} ${getProgressClassNameByProgress(progress)}`;
+
+function DonationCard({
+   title,
+   description,
+   donationTarget,
+   amount,
+   isOpen,
+   onShowMore,
+}: DonationCardProps) {
+   const progress = (amount / donationTarget) * 100;
+
    return (
-      <article className={styles.donationCard}>
-         <header>
-            <h3>{title}</h3>
+      <article className={styles.donationCard} onClick={onShowMore}>
+         <header className={styles.header}>
+            <h2 className={styles.title}>{title}</h2>
+            <button className={getButtonClassNames(progress)}>
+               Donate Now!
+            </button>
          </header>
-         <section>
-            <p>{description}</p>
-            <footer>
-               <p>Target: {donationTarget}</p>
-            </footer>
-         </section>
+         {isOpen && (
+            <section className={styles.description}>{description}</section>
+         )}
+         <footer>
+            <section className={styles.targetInfo}>
+               <div className={styles.progressBar}>
+                  <div
+                     className={getDonationCardProgressClassNames(progress)}
+                     style={{
+                        width: `${progress}%`,
+                        pointerEvents: 'none',
+                     }}>
+                     {progress.toFixed(0)}%
+                  </div>
+               </div>
+               <p className={styles.target}>
+                  ${donationTarget.toLocaleString()}
+               </p>
+            </section>
+            <button className={styles.button} onClick={onShowMore}>
+               {isOpen ? 'Show Less' : 'Show More'}
+            </button>
```

Comment:
Maybe get an advice from one of the UX if this button is necessary, if you can just click on the whole card to show more, and also the cards give you signs that it is clickable.

Or maybe the card shouldn't be clickable?

Its a user experience question, so ask one about it.
You should do it now because it is not only about the design but about the UX so it might affect the functionality that you added here.

Reply:
Discussed